### ISSUE TITLE
💅 Invalid unnecessary constructor on abstract class that forwards arguments to base

### ISSUE THREAD

**Issue by junlarsen** [biomejs/biome#2682]
### Environment information

```bash
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-kitty"
  JS_RUNTIME_VERSION:           "v20.11.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Linter:
  Recommended:                  false
  All:                          false
  Rules:                        unset

Workspace:
  Open Documents:               0
```

### Rule name

lint/complexity/noUselessConstructor

### Playground link

https://biomejs.dev/playground/?code=YQBiAHMAdAByAGEAYwB0ACAAYwBsAGEAcwBzACAAQgBhAHMAZQAgAHsACgAgACAAcAByAG8AdABlAGMAdABlAGQAIABjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABwAHIAbwB0AGUAYwB0AGUAZAAgAHIAZQBhAGQAbwBuAGwAeQAgAG4AYQBtAGUAOgAgAHMAdAByAGkAbgBnACkAIAB7AH0ACgB9AAoACgBjAGwAYQBzAHMAIABHAG8AbwBkAFAAYQByAGUAbgB0ACAAZQB4AHQAZQBuAGQAcwAgAEIAYQBzAGUAIAB7AAoAIAAgAHAAdQBiAGwAaQBjACAAYwBvAG4AcwB0AHIAdQBjAHQAbwByACgAKQAgAHsACgAgACAAIAAgAHMAdQBwAGUAcgAoACIAaABlAGwAbABvACAAdwBvAHIAbABkACIAKQAKACAAIAB9AAoAfQAKAAoAYwBsAGEAcwBzACAAQgBpAG8AbQBlAEQAbwBlAHMATgBvAHQATABpAGsAZQAgAGUAeAB0AGUAbgBkAHMAIABCAGEAcwBlACAAewAKACAAIABwAHUAYgBsAGkAYwAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoAHYAYQBsAHUAZQA6ACAAcwB0AHIAaQBuAGcAKQAgAHsACgAgACAAIAAgAHMAdQBwAGUAcgAoAHYAYQBsAHUAZQApAAoAIAAgAH0ACgB9AAoACgBjAGwAYQBzAHMAIABCAGkAbwBtAGUATABpAGsAZQBzACAAZQB4AHQAZQBuAGQAcwAgAEIAYQBzAGUAIAB7AH0A

### Expected result

Hi Biome team. Thanks for your wonderful work on Biome, it's truly refreshing to see such a good tool!

Biome incorrectly reports that the constructor of `BiomeDoesNotLike` is unnecessary. That is wrong, because since the constructor of `Base` is protected, fully removing the public constructor of `BiomeDoesNotLike` makes `BiomeDoesNotLike` uninstantiatable.

Instead, the `/lint/complexity/noUselessConstructor` rule should take visibility into account too.

A demonstration of how the TS compiler refuses the `BiomeLikes` constructor call is shown here: https://www.typescriptlang.org/play/?#code/IYIwzgLgTsDGEAJYBthjAgQmgpgg3gFAIIAOUA9hDvDgCZIUB2kUArvBVABTlU3UGUHMDrNkATwRNgAWxwAuBKwCWTAOYBKAgF9CewijQYA4hQp0ACsGFNEOAB7UmdDNjB4iJUmxDIVsIws0BwQXNzaXiTKbKQ4PABEABY4yMgUCADuXMh0CZrECHoGRuhYKhTyACIUOGAAclQAMioA1niOzq5YuASFPn4BQayh4QBuwMhsisrQalp90TFxPBNTOAUkxYSGqGWYFfIt7RidOC5uvfgGTDiZ5ZU4NXWNEMc43AkAjABMAMz5Qi3e5mCzWWwQCKEAD00KyHQccXgCHilCgCBSwgANAgQDRgGwPD06AdHgg1JiVBAMBAUsMQpx0QAzSiyGFw9w4HGZJIBJLkjB8ai0OgAOgQADEuAhCQgwri8KBkHgwkC7g8jm06p9fgDNEA

In other words, the suggestion made by the lint alters the meaning of the program.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by junlarsen** [comment]
For what it's worth, I wouldn't mind trying to contribute a fix once triaged and verified :smiley:

**Comment by Conaclos** [comment]
Hi! Unfortunately we are not currently able to retrieve information about the inherited class. Thus, it is not possible to know that the constructor of `Base` is protected.

**Comment by junlarsen** [comment]
Do you think it would be a reasonable thing to add a note to the diagnostic message in that case?

**Comment by Conaclos** [comment]
Closing because it is a duplicate of #987

**Comment by Conaclos** [comment]
Closing because it is a duplicate of https://github.com/biomejs/biome/issues/987

**Comment by junlarsen** [comment]
Apologies, I didn't see the issue. I'll comment over there
